### PR TITLE
Document `TypeInner::BindingArray`.

### PR DESCRIPTION
**Pull request by jimblandy**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -698,7 +698,23 @@ pub enum TypeInner {
     },
     /// Can be used to sample values from images.
     Sampler { comparison: bool },
-    /// Array of bindings
+    /// Array of bindings.
+    ///
+    /// A binding array can be an array of textures or samplers.
```
Comment (cwfitzgerald) [inline src/lib.rs:703]: They can also be an array of either uniform or storage data.

**Review by cwfitzgerald — APPROVED** [review]
Good outside of one comment

**Comment by jimblandy** [comment]
Okay, this is basically a complete rewrite, but I think it's more useful.

File: src/lib.rs [hunk 2]
```diff
@@ -698,7 +698,48 @@ pub enum TypeInner {
...
+    ///
+    /// A `BindingArray` represents an array where each element draws its value
+    /// from a separate bound resource. The array's element type `base` may be
+    /// [`Image`], [`Sampler`], or any type that would be permitted for a global
+    /// in the [`Uniform`] or [`Storage`] address space. Only global variables
+    /// may be binding arrays; on the host side, their values are provided by
+    /// [`TextureViewArray`], [`SamplerArray`], or [`BufferArray`]
+    /// bindings.
+    ///
+    /// Since each element comes from a distinct binding, a binding array of
```
Comment (cwfitzgerald) [inline src/lib.rs:712]:
```suggestion
    /// Since each element comes from a distinct resource, a binding array of
```

**Review by cwfitzgerald — APPROVED** [review]
One nit, but otherwise lgtm

**Pull request merged**